iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorNonLinearModelImpl.java

```diff
@@ -116,17 +127,40 @@ public int getMaximumSectionCount() {
     }
 
     @Override
-    public double getB(int sectionNum) {
-        return Optional.ofNullable(sections.get(sectionNum))
+    public double getCurrentB(int currentSectionCount) {
+        return sections.entrySet().stream()
+                .filter(e -> e.getKey() <= currentSectionCount)
+                .mapToDouble(e -> e.getValue().getB())
+                .sum();
+    }
+
+    @Override
+    public double getCurrentG(int currentSectionCount) {
+        return sections.entrySet().stream()
+                .filter(e -> e.getKey() <= currentSectionCount)
+                .mapToDouble(e -> e.getValue().getG())
+                .filter(g -> !Double.isNaN(g))
+                .sum();
+    }
+
+    @Override
+    public double getB(int sectionIndex) {
+        return Optional.ofNullable(sections.get(sectionIndex))
                 .map(SectionImpl::getB)
-                .orElseThrow(() -> new PowsyblException(invalidSectionNumberMessage(sectionNum, "susceptance")));
+                .orElseThrow(() -> new PowsyblException(invalidSectionNumberMessage(sectionIndex, "susceptance")));
     }
 
     @Override
-    public double getG(int sectionNum) {
-        return Optional.ofNullable(sections.get(sectionNum))
-                .map(SectionImpl::getG)
-                .orElseThrow(() -> new PowsyblException(invalidSectionNumberMessage(sectionNum, "conductance")));
+    public double getG(int sectionIndex) {
+        return Optional.ofNullable(sections.get(sectionIndex))
+                .map(section -> {
+                    double g = section.getG();
+                    if (!Double.isNaN(g)) {
+                        return g;
+                    }
+                    return 0.0;
+                })
+                .orElseThrow(() -> new PowsyblException(invalidSectionNumberMessage(sectionIndex, "conductance")));
```

Comment:
Sometimes, using the functional API is not a good idea. The code could look simpler using a procedural implementation

Reply:
Done